readinglist/tests/test_views_article.py

```diff
@@ -53,8 +53,14 @@ def test_mark_by_and_on_are_set_to_none_if_unread_is_true(self):
             'marked_read_by': 'FxOS',
             'marked_read_on': 1234}
         resp = self.app.patch_json(self.url, mark_read, headers=self.headers)
-        self.assertIsNotNone(resp.json['marked_read_by'])
-        self.assertIsNotNone(resp.json['marked_read_on'])
+        body = resp.json
+        self.assertEqual(body['marked_read_by'], mark_read['marked_read_by'])
+        self.assertEqual(body['marked_read_on'], mark_read['marked_read_on'])
+
+        resp = self.app.get(self.url, headers=self.headers)
+        body = resp.json
+        self.assertEqual(body['marked_read_by'], mark_read['marked_read_by'])
+        self.assertEqual(body['marked_read_on'], mark_read['marked_read_on'])
 
```

Comment:
I don't understand what you are trying to do here...

The test says *"mark_by_and_on_are_set_to_none"* and the what the test currently asserts is "it is not none before", then change unread, then "it is none after".

If you need to assert other things, then start a new test method with an explicit spec :)